xarray/core/coordinates.py

```diff
@@ -209,13 +216,38 @@ def _to_dataset(self, shallow_copy=True):
     def to_dataset(self):
         return self._to_dataset()
 
+    def __setitem__(self, key, value):
+        if key in self._data._level_coords:
+            raise ValueError("cannot replace MultiIndex level %r, replace %r "
+                             "coordinate instead"
+                             % (key, self._data._level_coords[key]))
+        return super(DataArrayCoordinates, self).__setitem__(key, value)
+
     def __delitem__(self, key):
         if key in self.dims:
             raise ValueError('cannot delete a coordinate corresponding to a '
                              'DataArray dimension')
         del self._data._coords[key]
 
 
+class DataArrayLevelCoordinates(AbstractCoordinates):
+    """Dictionary like container for DataArray MultiIndex level coordinates.
```

Comment:
Probably good to clarify "Used for attribute style lookup. Not returned directly by any public methods."